File: src/main/resources/reverse_number.rb

puts "Enter a number:"
n = gets.chomp.to_i

reversed = 0

# Reverse the number using a loop
while n != 0
  last_digit = n % 10
  reversed = reversed * 10 + last_digit
  n = n / 10
end

# Print the reversed number
puts "The reversed number is: #{reversed}"

File: src/main/resources/reverse_number.kt
import java.util.Scanner

fun main() {
  val scanner = Scanner(System.`in`)

  print("Enter a number: ")
  var n = scanner.nextInt()

  var reversed = 0

  // Reverse the number using a loop
  while (n != 0) {
    val lastDigit = n % 10
    reversed = reversed * 10 + lastDigit
    n /= 10
  }

  // Print the reversed number
  println("The reversed number is: " + reversed)
}